### PR TITLE
[TAS-2095] ✨ Add stripe product and price on new book listing

### PR DESCRIPTION
**Pull request by williamchong**
- create product and price on listing create
- recreate price on price change
- update product with iscn info when a listing update is triggered

### CODE REVIEW

**Review by nwingt — APPROVED** [review]

**Comment by williamchong** [comment]
@nwingt changed to use 1 stripe product for each book edition, please review again

File: src/util/api/likernft/book/index.ts [hunk 1]
```diff
@@ -9,15 +9,23 @@ import {
...
 import { sleep } from '../../../misc';
+import stripe from '../../../stripe';
+import { parseImageURLFromMetadata } from '../metadata';
 
 export const MIN_BOOK_PRICE_DECIMAL = 90; // 0.90 USD
 export const NFT_BOOK_TEXT_LOCALES = ['en', 'zh'];
 export const NFT_BOOK_TEXT_DEFAULT_LOCALE = NFT_BOOK_TEXT_LOCALES[0];
 
+export function getLocalizedTextWithFallback(field, locale) {
+  return field[locale] || field[NFT_BOOK_TEXT_DEFAULT_LOCALE] || undefined;
```
Comment (nwingt) [inline src/util/api/likernft/book/index.ts]: Use empty string for fallback?
```suggestion
  return field[locale] || field[NFT_BOOK_TEXT_DEFAULT_LOCALE] || '';
```

File: src/util/api/likernft/book/index.ts [hunk 2]
```diff
@@ -64,6 +72,42 @@ export function formatShippingRateInfo(shippingRate) {
...
+    name,
+    description,
+    iscnIdPrefix,
+    image,
+  } = bookInfo;
+  const images: string[] = [];
+  if (image) images.push(parseImageURLFromMetadata(image));
+  // if (thumbnailUrl) images.push(parseImageURLFromMetadata(thumbnailUrl));
+  const stripeProduct = await stripe.products.create({
+    name: `${name} - ${getLocalizedTextWithFallback(price.name, 'zh') || ''}`,
```
Comment (nwingt) [inline src/util/api/likernft/book/index.ts]:
```suggestion
    name: [name, getLocalizedTextWithFallback(price.name, 'zh')].filter(Boolean).join(' - '),
```

File: src/util/api/likernft/book/index.ts [hunk 3]
```diff
@@ -64,6 +72,42 @@ export function formatShippingRateInfo(shippingRate) {
...
+    description: `${getLocalizedTextWithFallback(price.description, 'zh') || ''}\n${description || ''}`,
```
Comment (nwingt) [inline src/util/api/likernft/book/index.ts]:
```suggestion
    description: [getLocalizedTextWithFallback(price.description, 'zh'), description).filter(Boolean).join('\n'),
```

File: src/util/api/likernft/book/index.ts [hunk 4]
```diff
@@ -151,6 +208,63 @@ export async function newNftBookInfo(classId, data, apiWalletOwnedNFTIds: string
...
+  if (image) payload.image = image;
+  await likeNFTBookCollection.doc(classId).update(payload);
+  await Promise.all(prices.map(async (p) => {
+    if (p.stripeProductId) {
+      const images: string[] = [];
+      if (image) images.push(parseImageURLFromMetadata(image));
+      if (thumbnailUrl) images.push(parseImageURLFromMetadata(thumbnailUrl));
+      await stripe.products.update(p.stripeProductId, {
+        name: `${name} - ${getLocalizedTextWithFallback(p.name, 'zh') || ''}`,
+        description: `${getLocalizedTextWithFallback(p.description, 'zh') || ''}\n${description || ''}`,
```
Comment (nwingt) [inline src/util/api/likernft/book/index.ts]:
```suggestion
        name: [name, getLocalizedTextWithFallback(p.name, 'zh')].filter(Boolean).join(' - '),
        description: [getLocalizedTextWithFallback(p.description, 'zh'), description].filter(Boolean).join('\n'),
```

**Review by nwingt — APPROVED** [review]

**Pull request merged**